skyportal/utils/gcn.py

```diff
@@ -0,0 +1,254 @@
+            yield value.lower()
+
+    # Get LIGO/Virgo source classification, if present.
+    classifications = [
+        (float(elem.attrib['value']), elem.attrib['name'])
+        for elem in root.iterfind("./What/Group[@type='Classification']/Param")
+    ]
+    if classifications:
+        _, classification = max(classifications)
+        yield classification
+
+    search = root.find("./What/Param[@name='Search']")
+    if search is not None:
+        yield search.attrib['value']
+
+
+def get_skymap(root, gcn_notice):
+    mission = urlparse(root.attrib['ivorn']).path.lstrip('/')
+
+    # Try Fermi GBM convention
+    if gcn_notice.notice_type == gcn.NoticeType.FERMI_GBM_FIN_POS:
+        url = root.find("./What/Param[@name='LocationMap_URL']").attrib['value']
+        url = url.replace('http://', 'https://')
+        url = url.replace('_locplot_', '_healpix_')
+        url = url.replace('.png', '.fit')
+        return from_url(url)
+
+    # Try Fermi GBM **subthreshold** convention. Stupid, stupid, stupid!!
+    if gcn_notice.notice_type == gcn.NoticeType.FERMI_GBM_SUBTHRESH:
+        url = root.find("./What/Param[@name='HealPix_URL']").attrib['value']
+        return from_url(url)
+
+    # Try LVC convention
+    skymap = root.find("./What/Group[@type='GW_SKYMAP']")
+    if skymap is not None:
+        children = skymap.getchildren()
+        for child in children:
+            if child.attrib['name'] == 'skymap_fits':
+                url = child.attrib['value']
+                break
+
+        return from_url(url)
+
+    retraction = root.find("./What/Param[@name='Retraction']")
+    if retraction is not None:
+        retraction = int(retraction.attrib['value'])
+        if retraction == 1:
+            return None
+
+    # Try error cone
+    loc = root.find('./WhereWhen/ObsDataLocation/ObservationLocation')
+    if loc is None:
+        return None
+
+    ra = loc.find('./AstroCoords/Position2D/Value2/C1')
+    dec = loc.find('./AstroCoords/Position2D/Value2/C2')
+    error = loc.find('./AstroCoords/Position2D/Error2Radius')
+
+    if None in (ra, dec, error):
+        return None
+
+    ra, dec, error = float(ra.text), float(dec.text), float(error.text)
+
+    # Apparently, all experiments *except* AMON report a 1-sigma error radius.
+    # AMON reports a 90% radius, so for AMON, we have to convert.
+    if mission == 'AMON':
+        error /= scipy.stats.chi(df=2).ppf(0.95)
+
+    return from_cone(ra, dec, error)
+
+
+def from_cone(ra, dec, error):
+    localization_name = "%.5f_%.5f_%.5f" % (ra, dec, error)
+
+    center = SkyCoord(ra * u.deg, dec * u.deg)
+    radius = error * u.deg
+
+    # Determine resolution such that there are at least
+    # 16 pixels across the error radius.
+    hpx = HEALPix(
+        pixel_resolution_to_nside(radius / 16, round='up'), 'nested', frame=ICRS()
+    )
+
+    # Find all pixels in the 4-sigma error circle.
+    ipix = hpx.cone_search_skycoord(center, 4 * radius)
```

Comment:
Are values like the `16 pixels across the error radius` and `4-sigma error circle` standards that are unlikely to change? Just wondering if it would be better to make these configurable via the config.yaml files.

Reply:
Yeah I don't think so. We haven't changed that since Fermi program started at start of ZTF I.